src/greeter/GreeterApp.cpp

```diff
@@ -134,6 +145,17 @@ namespace SDDM {
         connect(this, &GreeterApp::primaryScreenChanged, this, [this](QScreen *) {
             activatePrimary();
         });
+
+        // start timer to reset the form if user is idle
+        int formResetTime = SDDM::mainConfig.FormResetTime.get();
+        if (formResetTime > 0) {
+            if (m_metadata->formResetSupport()) {
+                connect(&m_formResetTimer, &QTimer::timeout, m_proxy, &GreeterProxy::formResetAction);
```

Comment:
you can connect a signal to a signal directly

i.e

 +                connect(&m_formResetTimer, &QTimer::timeout, m_proxy, &GreeterProxy::formReset);